components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImplTest.java

```diff
@@ -0,0 +1,336 @@
+@WithCarbonHome
+@WithH2Database(files = {"dbscripts/h2.sql"})
+@WithRegistry
+@WithRealmService(injectToSingletons = {IdentityCoreServiceDataHolder.class})
+public class ActionManagementServiceImplTest {
+
+    private static final Map<String, BasicDataSource> dataSourceMap = new HashMap<>();
+    private MockedStatic<IdentityDatabaseUtil> identityDatabaseUtil;
+    private static final String DB_NAME = "action_mgt";
+    private static final String ACCESS_TOKEN = "6e47f1f7-bd29-41e9-b5dc-e9dd70ac22b7";
+    private Action action;
+    private String tenantDomain;
+    private ActionManagementService serviceImpl;
+
+    @BeforeClass
+    public void setUpClass() throws Exception {
+
+        serviceImpl = ActionManagementServiceImpl.getInstance();
+        tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        initiateH2Database(getFilePath());
+    }
+
+    @BeforeMethod
+    public void setUp() throws SecretManagementException {
+
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        SecretManagerImpl secretManager = mock(SecretManagerImpl.class);
+        SecretType secretType = mock(SecretType.class);
+        ActionMgtServiceComponentHolder.getInstance().setSecretManager(secretManager);
+        when(secretType.getId()).thenReturn("secretId");
+        when(secretManager.getSecretType(any())).thenReturn(secretType);
+        mockDBConnection();
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        identityDatabaseUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+
+        closeH2Database();
+    }
+
+    @Test(priority = 1)
+    public void testAddAction() throws ActionMgtException {
+
+        Action creatingAction = buildMockAction(
+                "PreIssueAccessToken",
+                "To configure PreIssueAccessToken",
+                "https://example.com",
+                buildMockBasicAuthentication("admin", "admin"));
+        action = serviceImpl.addAction(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), creatingAction,
+                tenantDomain);
+        Assert.assertNotNull(action.getId());
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertEquals(creatingAction.getDescription(), action.getDescription());
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Assert.assertEquals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType(),
+                action.getType().getActionType());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getProperties().size(),
+                action.getEndpoint().getAuthentication().getProperties().size());
+    }
```

Comment:
Also let's use, 
action.getEndpoint().getAuthentication().getPropertiesWithSecretReferences() to get the references for the secrets and verify the property values in the response.

By this we can assert that the secret values are encrypted.

Also have API_KEY authentication type in later method and verify that the header property is not encrypted.

And use,
action.getEndpoint().getAuthentication().getPropertiesWithDecryptedValues() and verify whether the values are encrypted propertly.

Reply:
This is sorted as per the code